src/core/shapes/ConicalFrustum.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2010-2019 The ESPResSo project
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <boost/algorithm/clamp.hpp>
+
+#include "ConicalFrustum.hpp"
+#include <utils/Vector.hpp>
+#include <utils/math/coordinate_transformation.hpp>
+
+/*
+ *
+   -------->r1
```

Comment:
I think you should move this beautiful ascii art to the docstring of the class. Also you should add a high-level docstring to the class which describes the shape (e.g. like the python doc string).